node/builder_chain.go

```diff
@@ -149,6 +150,8 @@ var ChainNode = Options(
 		Override(HandleIncomingMessagesKey, modules.HandleIncomingMessages),
 		Override(HandleIncomingBlocksKey, modules.HandleIncomingBlocks),
 	),
+
+	If(build.F3Enabled, Override(new(*lf3.F3), lf3.New)),
```

Comment:
This will be instantiated even if the node is not a Miner. But AFAICT, only Miners participating in F3 need this. So why not move it to `builder_miner` ?